lesson15/bilge/script.js

```diff
@@ -0,0 +1,182 @@
+
+// 6. Check if String Contains Substring
+// Verify if a URL contains "https".
+const url = 'HTTPS://example.com';
+const protocol = 'https';
+
+console.log('Ex. 6:');
+console.log(url.toLowerCase().includes(protocol));
+
+// 7. Find Character at Index
+// Find the initials of the name.
+const firstName = 'D John Doe D ';
+
+console.log('Ex. 7:');
+console.log(firstName.indexOf('D', 12), firstName.indexOf('J'));
+
+if (firstName.indexOf('Jane') === -1) {
+	console.log("Name isn't Jane");
+}
+
+// 8. Find Index of Substring (H)
+// Find the position of a word 'JavaScript' in a sentence.
+const sentenceWithAWordToFind =
+	"The object-oriented nature of JavaScript is important to understand if you want to go further with your knowledge of the language and write more efficient code, therefore we've provided this module to help you.";
+
+console.log('Ex. 8:');
+console.log(sentenceWithAWordToFind.indexOf('JavaScript'));
+
+// 9. Extract a Substring
+// Extract the domain from an email address.
+const email = 'user-name@google.com';
+const startIndexOfDomain = email.indexOf('@') + 1;
+const domain = email.substring(startIndexOfDomain);
+
+
+console.log('Ex. 9:');
+console.log(domain);
+
+// 10. Replace Substring
+// Censor (replace 'bad' with 'good') a word in a sentence.
+const sentence = 'This is a bad example bad.';
+
+console.log('Ex. 10:');
+console.log(sentence.replace('bad', 'good'));
+
+// 11. Repeat a String (H)
+// Create a visual separator for console output. Expected result: "--------------------"
+const expression = 'hey';
+
+console.log('Ex. 11:');
+console.log(expression.repeat(3));
+
+// 12. Check if String Starts With (H)
+// Check if a file name starts with a specific prefix. Return true or false.
+const fileName = 'report_2024.pdf';
+
+console.log('Ex. 12:');
+console.log(fileName.startsWith('report'));
+
+// 13. Check if String Ends With (H)
+// Check if a file name ends with a specific extension. Return true or false.
+const extension = 'abc.pdf';
+
+console.log('Ex. 13:');
+console.log(extension.endsWith('.pdf'));
+
+// 14. Pad Start of String (H)
+// Pad a number with zeros using padStart() function to ensure it has at least 5 digits.
+const number = '42';
+
+console.log('Ex. 14:');
+console.log(number.padStart(5, '0'));
+
+// 15. Pad End of String (H)
+// Pad a text with spaces to align it to the right. Expected result: "Align     "
+const text = 'Align';
+
+console.log('Ex. 15:');
+console.log(text.padEnd(20), text); // I added the text variable next to the padded string to show it's aligned to the side.
+
+// 16. Split a String (H)
+// Split a sentence into words.
+const sentence2 = 'Learning JavaScript is fun';
+const splittedVersion = sentence2.split(' ');
+
+console.log('Ex. 16:');
+console.log(splittedVersion);
+
+// 17. Convert String to Array of Characters (H)
+// Convert a string into an array of its characters. Use sentence2 variable from exercise above.
+const convertToArray =  Array.from(sentence2);
```

Comment:
Another way of solving this is: `sentence2.split('');`

Reply:
If we used split function, then it'd split the sentence into words I guess. Can it convert the sentence into array and maybe needs another line of code to convert it?

Reply:
If you use space in the split, then you will split sentence into words. If you use empty string in split, then you will split it into characters.
`sentence2.split(' '); // returns an array of words (if any)`

`sentence2.split(' '); // returns an array of characters (if any)`

Reply:
Ok I got it, thanks for clarification Anna! :)